### ISSUE TITLE
Destroying the HUB 

### ISSUE THREAD

**Issue by falahati** [mircearoata/SatisfactoryAreaActions#1]
Hey,

If you accidentally or purposefully destroy your HUB with this MOD, the base of the HUB stays and then can not be deleted as it crashes the game and you can't make another one since the game still believes that the HUB is there.

It would nice to ignore the HUB while clearing an area.

**Comment by mircearoata** [comment]
The mod does ignore the HUB, but it can't ignore the built in burners, MAM, and workbench. Can you send the crash log so I can check what went wrong, because last time I checked, it worked without crashing

**Comment by falahati** [comment]
Well, I don't know how to enable logging for the game. And the `CMD` window that opens while using MODS just closes after a while and I couldn't say if I found anything related to the crash. There is also a message box that opens that supposed to have crash details but it is just empty and there is only an "Ok" button that when clicked will close the game immediately.

I can only tell you that the HUB's base is there and is not removed. However, there is nothing on top of it. Here is a photo:
![SF1](https://user-images.githubusercontent.com/2479372/69756941-e7221000-1170-11ea-9574-0dd1efee731b.jpg)


If I choose the demolish tool (`F`) and as soon as I hover the HUB's base, the game will crash. As if it can't calculate what material I am going to get from demolishing this building. I suspect that there is a connection between the HUB's platform and the buildings that are a part of it and since all other buildings have deleted this connection is now corrupted.

**Comment by mircearoata** [comment]
The logs of the game are in %localappdata%\FactoryGame\Logs

Upload the one from just after the game crashed (here or pastebin)

**Comment by falahati** [comment]
Good to know.
Unfortunately, as I suspected, there is not much information there. The only related lines to the crash are the following:


> [2019.11.27-**20.11.26:790**][220]LogOnline: AFGGameSession::UpdateGameSession() seemed to work!
> [2019.11.27-**20.12.49:502**][ 46]LogWindows: Warning: CreateProc failed: The system cannot find the file specified. (0x00000002)
> [2019.11.27-20.12.49:502][ 46]LogWindows: Warning: URL: ../../../Engine/Binaries/Win64/CrashReportClient.exe "C:/Users/s_fal/AppData/Local/FactoryGame/Saved/Crashes/UE4CC-Windows-C3C44A404B75E9F81095379130FA9BF3_0004" -AppName=UE4-FactoryGame -CrashGUID=UE4CC-Windows-C3C44A404B75E9F81095379130FA9BF3_0004 -DebugSymbols=..\..\..\Engine\Intermediate\Symbols
> [2019.11.27-20.12.49:502][ 46]LogWindows: Could not start CrashReportClient.exe
> [2019.11.27-20.12.49:502][ 46]LogMemory: Platform Memory Stats for WindowsNoEditor
> [2019.11.27-20.12.49:502][ 46]LogMemory: Process Physical Memory: 4516.71 MB used, 5662.97 MB peak
> [2019.11.27-20.12.49:502][ 46]LogMemory: Process Virtual Memory: 8337.16 MB used, 8899.47 MB peak
> [2019.11.27-20.12.49:502][ 46]LogMemory: Physical Memory: 15203.45 MB used,  1115.00 MB free, 16318.45 MB total
> [2019.11.27-20.12.49:502][ 46]LogMemory: Virtual Memory: 13469.77 MB used,  134204256.00 MB free, 134217728.00 MB total
> [2019.11.27-20.12.49:502][ 46]LogWindows: Error: === Critical error: ===
> [2019.11.27-20.12.49:502][ 46]LogWindows: Error: 
> [2019.11.27-20.12.49:502][ 46]LogWindows: Error: 
> [2019.11.27-20.12.49:575][ 46]LogExit: Executing StaticShutdownAfterError
> [2019.11.27-20.12.50:968][ 46]LogWindows: FPlatformMisc::RequestExit(1)
> [2019.11.27-20.12.50:974][ 46]Log file closed, 11/27/19 23:42:50

Unfortunately, the crash dump also wasn't very helpful:

```
This dump file has an exception of interest stored in it.
The stored exception information can be accessed via .ecxr.
(aac.5e30): Access violation - code c0000005 (first/second chance not available)
ntdll!NtGetContextThread+0x14:
00007ffb`f56dde54 c3              ret
```
CallStack:

```
FactoryGame-Win64-Shipping  0x0000000058860000 + 8663d7  
FactoryGame-Win64-Shipping  0x0000000058860000 + 863ddf  
FactoryGame-Win64-Shipping  0x0000000058860000 + a68cd1  
FactoryGame-Win64-Shipping  0x0000000058860000 + cb1892  
FactoryGame-Win64-Shipping  0x0000000058860000 + e29a20  
FactoryGame-Win64-Shipping  0x0000000058860000 + 1bea4e4 
FactoryGame-Win64-Shipping  0x0000000058860000 + a652f8  
FactoryGame-Win64-Shipping  0x0000000058860000 + 860169  
FactoryGame-Win64-Shipping  0x0000000058860000 + 8662f5  
FactoryGame-Win64-Shipping  0x0000000058860000 + a4bfd4  
FactoryGame-Win64-Shipping  0x0000000058860000 + cb1892  
FactoryGame-Win64-Shipping  0x0000000058860000 + e0f37f  
FactoryGame-Win64-Shipping  0x0000000058860000 + e29d4f  
FactoryGame-Win64-Shipping  0x0000000058860000 + e0fa98  
FactoryGame-Win64-Shipping  0x0000000058860000 + e29d4f  
FactoryGame-Win64-Shipping  0x0000000058860000 + cb1892  
FactoryGame-Win64-Shipping  0x0000000058860000 + e29a20  
FactoryGame-Win64-Shipping  0x0000000058860000 + a49e33  
FactoryGame-Win64-Shipping  0x0000000058860000 + 8645cb  
FactoryGame-Win64-Shipping  0x0000000058860000 + 1bf401a 
FactoryGame-Win64-Shipping  0x0000000058860000 + 1bd5c0a 
FactoryGame-Win64-Shipping  0x0000000058860000 + 23b5cb5 
FactoryGame-Win64-Shipping  0x0000000058860000 + 23bbabd 
FactoryGame-Win64-Shipping  0x0000000058860000 + adf222  
FactoryGame-Win64-Shipping  0x0000000058860000 + adf543  
FactoryGame-Win64-Shipping  0x0000000058860000 + ae8b88  
FactoryGame-Win64-Shipping  0x0000000058860000 + 23da1f3 
FactoryGame-Win64-Shipping  0x0000000058860000 + 23dfcb2 
FactoryGame-Win64-Shipping  0x0000000058860000 + 1fe539b 
XINPUT1_3                   0x00000000cfc70000 + 1b3ff   
FactoryGame-Win64-Shipping  0x0000000058860000 + 1ed7cda 
FactoryGame-Win64-Shipping  0x0000000058860000 + 907e36  
FactoryGame-Win64-Shipping  0x0000000058860000 + 2f7f73  
FactoryGame-Win64-Shipping  0x0000000058860000 + 2fef6c  
FactoryGame-Win64-Shipping  0x0000000058860000 + 2fefca  
FactoryGame-Win64-Shipping  0x0000000058860000 + 30810e  
FactoryGame-Win64-Shipping  0x0000000058860000 + 30169d7 
KERNEL32                    0x00000000f47a0000 + 17bd4   
ntdll                       0x00000000f5640000 + 6ced1   
```


BTW, even tho the memory seems to be quite full; this is irrelevant to the crash as I had this issue with 32GB memory as well. But maybe it corrupts the crash dump or something. Not sure about that.

Unfortunately, It seems that I can't help much and if you also can't reproduce this probably this can't be fixed soon. I will report back if I found any new information or a workaround.

**Comment by falahati** [comment]
Ok, I tried opening my save game via "Satisfactory Save Editor" to remove the HUB completely with hopes to be able to create it again and I just saw something interesting:

![image](https://user-images.githubusercontent.com/2479372/69801311-2f324880-11ec-11ea-9b9c-ea84df8e96f3.png)

If you look closely there are two generators associated with this building (Burners?) and they both have empty targets meaning there is no object (or if there is, the pointer is invalid and therefore this app won't show it).
Simply removing these two empty references allows for dismantling of the HUB's platform and the crash goes away.

So you probably need to ignore Burners as well (or check if there is any burner as a part of the hub) or better, just update the HUB's properties when removing those.
